### DISCUSSION TITLE
Android Java

### DISCUSSION THREAD

**Discussion by hwy419** [jemshit/WalletConnect#3] [General]
Trying to implement this in Java Android DApp and am hanging up with the Scarlet.Builder(). Specifically, .addMessageAdapterFactory and .addStreamAdapterFactory. I don't exactly know how to create the Message and Stream Adapter objects and pass into the the methods for Scarlet Builder. I'm still new to Java/Kotlin, so it's probably something very simple.

```
Scarlet scarlet = new Scarlet.Builder()
.webSocketFactory(webSocketFactory)

// tried these variations
.addMessageAdapterFactory(GsonMessageAdapter.Factory(gson))
.addMessageAdapterFactory(GsonMessageAdapter().Factory(gson))
.addMessageAdapterFactory(new GsonMessageAdapter.Factory(gson))
// ----tried
                
.addStreamAdapterFactory(streamAdapterFactories)
.backoffStrategy(ExponentialBackoffStrategy(1000,8000))
.lifecycle(lifecycleRegistry)
.build();
```

Seems like this person is also having a similar challenge, but the answer there about creating a kotlin file and where to put the code is not quite clear enough to me:
https://github.com/Tinder/Scarlet/issues/12#issuecomment-410157314

Any help is appreciated. Perhaps you know how?

**Comment by jemshit** [comment]
Did you add [this](https://github.com/jemshit/WalletConnect/blob/c00c9481a9ce872c8376a0efd7bd2fd7816ef4b5/buildSrc/src/main/java/Dependencies.kt#L25) dependency ?

`com.tinder.scarlet:message-adapter-gson:XYZ`

**Reply by hwy419** [reply]
Yes. I took a step back and re-worked all my code. Still very new to java/kotlin/android but I'm making progress and overcame whatever as causing my problem. DApp and Wallet have a successful bridge now. Thank you again for your work here it is very useful and helpful!
